src/backend/distributed/connection/connection_management.c

```diff
@@ -1454,6 +1454,9 @@ AfterXactHostConnectionHandling(ConnectionHashEntry *entry, bool isCommit)
 		{
 			ShutdownConnection(connection);
 
+			/* remove from transactionlist before free-ing */
+			ResetRemoteTransaction(connection);
```

Comment:
Is it always safe to dlist_delete here? Or do we plan to skip

Reply:
Running valgrind right now to find out. It definitely needs the second commit in this PR too (which we discussed yesterday). Otherwise crashes occur.

Reply:
Second commit makes sense